lib/internal/modules/esm/hooks.js

```diff
@@ -174,7 +191,46 @@ class Hooks {
       const next = this.#chains.load[this.#chains.load.length - 1];
       ArrayPrototypePush(this.#chains.load, { __proto__: null, fn: load, url, next });
     }
-    return initialize?.(data);
+
+    ArrayPrototypePush(this.#loaderInstances, instance);
+    return initialize?.(data, { __proto__: null, id: instance.id });
+  }
+
+  #removeFromChain(chain, target) {
+    for (let i = 0; i < chain.length; ++i) {
+      if (target === chain[i+1]?.fn) {
+        chain.splice(i+1, 1);
+      }
+      if (chain[i].next) {
+        chain[i].next = chain[i+1];
+      }
+    }
+  }
+
+  removeCustomLoader(id) {
+    const index = ArrayPrototypeFindIndex(this.#loaderInstances, (target) => {
+      return target.id === id;
+    });
+    if (index < 0) {
+      return false;
+    }
+    const instance = this.#loaderInstances[index];
```

Comment:
Since we are using `index` to just get the element, we can use `ArrayPrototypeFind` instead of index.